src/Core/ETW/CodeMarkers.cs

```diff
@@ -179,14 +179,21 @@ private void Write(int id)
         }
 
         [NonEvent]
-        private void Write(int id, params string[] texts)
+        private void Write(int id, string text)
         {
             if (IsEnabled())
             {
-                WriteEvent(id, texts);
+                WriteEvent(id, text);
```

Comment:
Why didn't it work before? Is it because the params array got passed as an array as well? So it ended up being array of array?

I'm guessing that this is because the base method accepts `params object[]`. What if we were to cast our array to object before sending?
I.e. `WriteEvent(texts.Cast<Object>().ToArray())`. If it works then we can keep our params method.

Reply:
Yes, that's probably what was happening. We don't need to cast the array, we just need to change our overload to accept `params object[]`.